source/fundamentals/logging.txt

```diff
@@ -0,0 +1,273 @@
+           <version>1.2.3</version>
+         </dependency>
+
+      Once you have installed the above dependencies, try connecting to a
+      MongoDB instance and retrieving a document with the following code.
+
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create(<my uri>);
+         MongoDatabase database = mongoClient.getDatabase(<my database>);
+         MongoCollection<Document> collection = database.getCollection(<my collection>);
+         collection.find().first();
+
+      You should see output resembling the following:
+
+      .. code-block:: none
+
+         ...
+         1156 [main] DEBUG o.m.driver.protocol.command - Sending command '{"find": "<your collection>", "filter": {}, "limit": 1, "singleBatch": true, "$db": "<your db>", "lsid": {"id": {"$binary": {"base64": "<server session>", "subType": "04"}}}}' with request id 13 to database <your db> on connection <your connection> to server <your server>
+         1173 [main] DEBUG o.m.driver.protocol.command - Execution of command with request id 13 completed successfully in 30.01 ms on connection <your connection> to server <your server>
+   
+      For more information on Logback, see the
+      `Logback manual <http://logback.qos.ch/manual/>`__.
+   
+   .. tab:: Log4j2
+      :tabid: Log4j2-binding
+   
+      If you add these dependencies to your ``pom.xml`` file, the MongoDB Java
+      driver uses the Log4j2 framework to output your logs.
+
+      .. code-block:: xml      
+      
+         <!-- SLF4J API -->
+         <dependency>
+           <groupId>org.slf4j</groupId>
+           <artifactId>slf4j-api</artifactId>
+           <version>1.7.30</version>
+         </dependency>
+         <!-- SLF4J-Log4j2 binding -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-slf4j-impl</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+         <!-- Log4j2 core functionality -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-core</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+
+      Once you have installed the above dependencies, try logging an error.
+
+      .. code-block:: java
+
+         import org.slf4j.Logger;
+         import org.slf4j.LoggerFactory;
+
+         ...
+
+         Logger logger = LoggerFactory.getLogger(App.class);
+         logger.error("Logging an Error");
+
+      You should see output resembling the following:
+
+      .. code-block:: none
+
+         12:35:00.438 [main] ERROR <my package path> - Logging an Error
+
+      For more information on Log4j2, see the
+      `Log4j2 manual <https://logging.apache.org/log4j/2.x/manual/>`__.
+
+Configure Your Logger
+---------------------
+
+To configure your logger, you must configure the logging framework bound to
+SLF4J.
+
+The example below shows how to configure your logger's log level.
```

Comment:
What's the difference between `logger` and `logging framework` here? Do they mean different things, or could they be seen as the same thing by the user?

Also, the example shows how to configure the log level. If that's the task, should it be the heading for this section?

Reply:
Good question. I tried to use "logger" to refer to the series of interconnected packages that allow the MongoDB Java Driver to output logs (`slf4j-api` + binding + logging framework), where as "logging framework" only refers to the package bound to `slf4j-api` that implements logging functionality. I agree that "logger" and "logging framework" are hard to distinguish between, but I am struggling to come up with better terms. I will continue to try and think of a clearer way to phrase this, and any suggestions are appreciated. 

I tried to use "configuring log level" as an example for showing how to configure your logger (use your logging framework's native configuration system). The concrete example tries to show how you can link logging framework documentation to your MongoDB Java Project. I can see how this is not clear as the page is written.